core/src/main/scala/org/apache/spark/rdd/UnionRDD.scala

```diff
@@ -62,7 +66,23 @@ class UnionRDD[T: ClassTag](
     var rdds: Seq[RDD[T]])
   extends RDD[T](sc, Nil) {  // Nil since we implement getDependencies
 
+  // Evaluate partitions in parallel. Partitions of each rdd will be cached by the `partitions`
+  // val in `RDD`.
+  private lazy val evaluatePartitions: Unit = {
+    val threshold = conf.getInt("spark.rdd.parallelListingThreshold", 10)
+    Preconditions.checkArgument(threshold > 0,
+      "spark.rdd.parallelListingThreshold must be positive: %s", threshold.toString)
+    if (rdds.length > threshold) {
+      val parArray = rdds.toParArray
+      parArray.tasksupport = new ForkJoinTaskSupport(new ForkJoinPool(threshold))
+      parArray.foreach(_.partitions)
+    } else {
+      rdds.foreach(_.partitions)
+    }
+  }
+
```

Comment:
I am not very comfortable with this lazy val. It is not obvious what it is doing and there is no way to disable it.